### PR TITLE
[14.0] l10n_it_reverse_charge supporting "with_supplier_self_invoice" for e-invoices + link supplier self invoice to e-invoice 

### PR DESCRIPTION
**Pull request by dcorio**
Porting del codice per v14

### CODE REVIEW

**Comment by dcorio** [comment]
...non ricordo come si mette il WIP :-)

**Comment by eLBati** [comment]
@dcorio intendi per metterla in draft? Sotto i revisori ci dovrebbe essere
image

**Comment by dcorio** [comment]
@eLBati grazie! trovato

File: l10n_it_reverse_charge/models/account_rc_type.py [hunk 1]
```diff
@@ -25,10 +30,36 @@ class AccountRCTypeTax(models.Model):
...
             "unique (rc_type_id,purchase_tax_id,sale_tax_id)",
-            "Tax mappings can be defined only once per rc type.",
-        )
+            'Tax mappings from Purchase Tax to Sale Tax '
+            'can be defined only once per Reverse Charge Type.'
+        ),
+        (
+            'original_purchase_sale_tax_uniq',
+            'unique (rc_type_id,'
+            'original_purchase_tax_id,purchase_tax_id,sale_tax_id)',
```
Comment (sergiocorato) [inline l10n_it_reverse_charge/models/account_rc_type.py:39]: Qui ci sono due apici di troppo

**Review by sergiocorato — CHANGES_REQUESTED** [review]
Puoi far girare pre-commit?

**Comment by sergiocorato** [comment]
check https://github.com/aion-tech/l10n-italy/pull/1

**Review by sergiocorato — APPROVED** [review]

**Review by simona-buccheri — APPROVED** [review]
LGTM

**Review by sergiocorato — CHANGES_REQUESTED** [review]
@dcorio puoi squashare i commit?

**Comment by sergiocorato** [comment]
@gabriele-baldessari

**Comment by dcorio** [comment]
@sergiocorato certo, procediamo

**Review by sergiocorato — APPROVED** [review]

**Comment by eLBati** [comment]
Mi pare manchi questo commit https://github.com/OCA/l10n-italy/pull/2790/commits/bc2dba202508e1a9b62301474db5a8ae576e7d50

**Comment by eLBati** [comment]
Incluso in https://github.com/OCA/l10n-italy/pull/3298

**Pull request merged**